### PR TITLE
chore: update grid behavior

### PR DESCRIPTION
**Pull request by oliverschuerch**

### CODE REVIEW

File: .changeset/fast-rocks-visit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@swisspost/design-system-styles': patch
+---
+
+Refactored the grid-system. Adjusted container paddings, added responsive gutter-widths and deprecated `rg` breakpoint.
```
Comment (imagoiq) [inline .changeset/fast-rocks-visit.md:5]:
```suggestion
Refactored the grid-system: adjusted container paddings, added responsive gutter-widths and deprecated `rg` breakpoint.
```

File: packages/documentation/src/stories/foundation/layout/grid-update/grid.stories.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                <div class="col-2"><div class="bg-container">11</div></div>
+                <div class="col-2"><div class="bg-container">12</div></div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `,
+};
```
Comment (imagoiq) [inline packages/documentation/src/stories/foundation/layout/grid-update/grid.stories.ts:86]: To be removed as said in the PR's description

File: packages/styles/src/components/grid.scss [hunk 3]
```diff
@@ -11,6 +11,95 @@
...
+    @if $infix == 'xs' {
+      .g-#{$postfix},
+      .gx-#{$postfix} {
+        --bs-gutter-x: #{$gutter};
+      }
+
+      // .g-#{$postfix},
+      // .gy-#{$postfix} {
+      //   --bs-gutter-y: #{$gutter};
+      // }
```
Comment (imagoiq) [inline packages/styles/src/components/grid.scss:84]:
```suggestion
```

File: packages/styles/src/components/grid.scss [hunk 4]
```diff
@@ -11,6 +11,95 @@
...
+      @include media-breakpoint-up($infix) {
+        .g-#{$infix}-#{$postfix},
+        .gx-#{$infix}-#{$postfix} {
+          --bs-gutter-x: #{$gutter};
+        }
+
+        // .g-#{$infix}-#{$postfix},
+        // .gy-#{$infix}-#{$postfix} {
+        //   --bs-gutter-y: #{$gutter};
+        // }
```
Comment (imagoiq) [inline packages/styles/src/components/grid.scss:95]:
```suggestion
```

File: packages/styles/src/components/grid.scss [hunk 5]
```diff
@@ -11,6 +11,95 @@
...
+      }
+    }
+  }
+}
+
+// legacy code
```
Comment (imagoiq) [inline packages/styles/src/components/grid.scss:101]: Should it be deprecated/removed?

**Review by imagoiq — APPROVED** [review]

File: packages/styles/src/components/grid.scss (hunk 5 above)
Comment (oliverschuerch) [reply packages/styles/src/components/grid.scss:93]: Not yet, there is another ticket to do this: [#2041](https://github.com/swisspost/design-system/issues/2041).

**Review by imagoiq — APPROVED** [review]

**Pull request merged**